File: sources/net.sf.j2s.core/test/src/com/falstad/circuit/Circuit.js
(function(){var P$=Clazz.newPackage("com.falstad.circuit"),I$=[];
var C$=Clazz.newClass(P$, "Circuit", null, 'a2s.Applet', 'java.awt.event.ComponentListener');
C$.ogf = null;

C$.$clinit$ = function() {Clazz.load(C$, 1);
}

Clazz.newMeth(C$, '$init0$', function () {
var c;if((c = C$.superclazz) && (c = c.$init0$))c.apply(this);
this.finished = false;
this.started = false;
}, 1);

Clazz.newMeth(C$, '$init$', function () {
this.finished = false;
this.started = false;
}, 1);

Clazz.newMeth(C$, 'destroyFrame', function () {
if (C$.ogf != null ) C$.ogf.dispose();
C$.ogf = null;
this.repaint();
this.finished = true;
});

Clazz.newMeth(C$, 'init', function () {
this.addComponentListener$java_awt_event_ComponentListener(this);
this.showFrame();
});

Clazz.newMeth(C$, 'main', function (args) {
C$.ogf = Clazz.new_((I$[0]||(I$[0]=Clazz.load('com.falstad.circuit.CirSim'))).c$$com_falstad_circuit_Circuit,[null]);
C$.ogf.init();
}, 1);

Clazz.newMeth(C$, 'showFrame', function () {
if (this.finished) {
this.repaint();
return;
}if (C$.ogf == null ) {
this.started = true;
C$.ogf = Clazz.new_((I$[0]||(I$[0]=Clazz.load('com.falstad.circuit.CirSim'))).c$$com_falstad_circuit_Circuit,[this]);
C$.ogf.init();
}C$.ogf.setVisible$Z(true);
this.repaint();
});

Clazz.newMeth(C$, 'hideFrame', function () {
if (this.finished) return;
C$.ogf.setVisible$Z(false);
this.repaint();
});

Clazz.newMeth(C$, 'toggleSwitch$I', function (x) {
C$.ogf.toggleSwitch$I(x);
});

Clazz.newMeth(C$, 'paint$java_awt_Graphics', function (g) {
C$.superclazz.prototype.paint$java_awt_Graphics.apply(this, [g]);
var s = "Applet is open in a separate window.";
if (C$.ogf != null  && !C$.ogf.isVisible() ) s = "Applet window is hidden.";
if (!this.started) s = "Applet is starting.";
 else if (C$.ogf == null  || this.finished ) s = "Applet is finished.";
 else if (C$.ogf != null  && C$.ogf.useFrame ) C$.ogf.triggerShow();
g.drawString$S$I$I(s, 10, 30);
});

Clazz.newMeth(C$, 'componentHidden$java_awt_event_ComponentEvent', function (e) {
});

Clazz.newMeth(C$, 'componentMoved$java_awt_event_ComponentEvent', function (e) {
});

Clazz.newMeth(C$, 'componentShown$java_awt_event_ComponentEvent', function (e) {
this.showFrame();
});

Clazz.newMeth(C$, 'componentResized$java_awt_event_ComponentEvent', function (e) {
if (C$.ogf != null ) C$.ogf.componentResized$java_awt_event_ComponentEvent(e);
});

Clazz.newMeth(C$, 'destroy', function () {
if (C$.ogf != null ) C$.ogf.dispose();
C$.ogf = null;
this.repaint();
});

Clazz.newMeth(C$);
})();
//Created 2017-12-17 19:28:18

File: sources/net.sf.j2s.java.core/src2/com/falstad/circuit/Circuit.java
package com.falstad.circuit;

// Circuit.java (c) 2005,2008 by Paul Falstad, www.falstad.com


// see CirSim.java for JavaScript implementation notes.

//web_Ready
//web_AppletName= Circuit
//web_Description= An electronic circuit simulator.  When the applet starts up you will see an animated schematic of a simple LRC circuit. The green color indicates positive voltage.  The gray color indicates ground.  A red color indicates negative voltage.  The moving yellow dots indicate current.
//web_JavaVersion= http://www.falstad.com/circuit/
//web_Location= com.falstad.Circuit.Circuit
//web_AppletImage= circuit.png
//web_Category= Physics - Electronics
//web_Date= $Date: 2017-01-01 00:08:18 -0600 (Sun, 01 Jan 2017) $
//web_Features= graphics, timer thread, pop-up menu, menubar, dialog, local file reading and writing, reflection

import java.awt.Graphics;
import java.awt.event.ComponentEvent;
import java.awt.event.ComponentListener;

import java.applet.Applet;

public class Circuit extends Applet implements ComponentListener {
	static CirSim ogf;
	boolean finished = false;

	void destroyFrame() {
		if (ogf != null)
			ogf.dispose();
		ogf = null;
		repaint();
		finished = true;
	}

	boolean started = false;

	public void init() {
		addComponentListener(this);
		showFrame();
	}

	public static void main(String args[]) {
		ogf = new CirSim(null);
		ogf.init();
	}

	public void showFrame() {
		if (finished) {
			repaint();
			return;
		}
		if (ogf == null) {
			started = true;
			ogf = new CirSim(this);
			ogf.init();
		}
		ogf.setVisible(true);
		repaint();
	}

	public void hideFrame() {
		if (finished)
			return;
		ogf.setVisible(false);
		repaint();
	}

	public void toggleSwitch(int x) {
		ogf.toggleSwitch(x);
	}

	public void paint(Graphics g) {
		super.paint(g); // paints background
		String s = "Applet is open in a separate window.";
		if (ogf != null && !ogf.isVisible())
			s = "Applet window is hidden.";
		if (!started)
			s = "Applet is starting.";
		else if (ogf == null || finished)
			s = "Applet is finished.";
		else if (ogf != null && ogf.useFrame)
			ogf.triggerShow();
		g.drawString(s, 10, 30);
	}

	public void componentHidden(ComponentEvent e) {
	}

	public void componentMoved(ComponentEvent e) {
	}

	public void componentShown(ComponentEvent e) {
		showFrame();
	}

	public void componentResized(ComponentEvent e) {
		if (ogf != null)
			ogf.componentResized(e);
	}

	public void destroy() {
		if (ogf != null)
			ogf.dispose();
		ogf = null;
		repaint();
	}
};
